src/components/EnterpriseContract/utils.tsx

```diff
@@ -44,8 +44,26 @@ export const getRuleStatus = (type: ENTERPRISE_CONTRACT_STATUS) => {
   }
 };
 
+/**
+ * This regex expect the logs from tekton results to be in this formay
+ *
+ * ```
+ * step-vulnerabilities :-
+ * Lorem Ipsum some logs
+ *
+ * step-report-json :-
+ * {"success":true,"components":[{"name":"devfile-sample-code-with-quarkus-1",<... ec report in JSON ...>,}]}
+ *
+ * ```
+ *
+ */
+const EC_REPORT_JSON_REGEX = /((?<=step-report-json\s*:-\s*)(\{.*?\})(?=\s*step-|$))/g;
+
 export const extractEcResultsFromTaskRunLogs = (logs: string): EnterpriseContractResult => {
-  const extractedLogs = logs.match(/(\[report-json\] ).+/g);
-  const json = JSON.parse(extractedLogs.map((l) => l.replace('[report-json] ', '')).join(''));
-  return json;
+  const extractedLogs = logs.match(EC_REPORT_JSON_REGEX);
+  try {
+    return JSON.parse(extractedLogs[0]);
+  } catch {
+    return null;
```

Comment:
Would not make sense to propagate this higher? And ignore it where it is used?

Reply:
Good catch, We should propagate it and show an error state to user rather than hiding it with an empty state.

Reply:
we also have the src/utils/enterprise-contract-utils.tsx  which is same with the one. We just need to keep one.